lib/matplotlib/collections.py

```diff
@@ -849,6 +849,24 @@ def set_alpha(self, alpha):
 
     set_alpha.__doc__ = artist.Artist._set_alpha_for_array.__doc__
 
+    def set_array(self, A):
+        """
+        Set the image array.
+
+        Parameters
+        ----------
+        A : array-like
+        """
+        # This method overrides the inherited
+        # cm.ScalarMappable.set_array method
+        self._A = cbook.safe_masked_invalid(A, copy=True)
+
+        if (self._A.dtype != np.uint8 and
+                not np.can_cast(self._A.dtype, float, "same_kind")):
+            raise TypeError("Image data of dtype {} cannot be converted to "
```

Comment:
You should make this check and raise before assigning to `self._A`.

Reply:
I'm not sure I understand, if one of the check fails, it will terminate anyway?
Alternatively, if this should be followed, should this be done here as well?
https://github.com/matplotlib/matplotlib/blob/ce2964854420fbb3e2a9519014e9e5b44824eb92/lib/matplotlib/image.py#L696-L701

Reply:
If you have set `self._A` then the internal state has been modified and the main purpose of the method call has been done!